File: Application/src/main/java/com/shareyourproxy/api/rx/event/RecyclerViewDatasetChangedEvent.java
package com.shareyourproxy.api.rx.event;


import com.shareyourproxy.app.adapter.BaseRecyclerView;
import com.shareyourproxy.app.adapter.BaseRecyclerViewAdapter;

/**
 * Created by Evan on 11/2/15.
 */
public class RecyclerViewDatasetChangedEvent {
    public final BaseRecyclerView.ViewState viewState;
    public final BaseRecyclerViewAdapter adapter;

    public RecyclerViewDatasetChangedEvent(BaseRecyclerViewAdapter adapter, BaseRecyclerView
        .ViewState viewState) {
        this.viewState = viewState;
        this.adapter = adapter;
    }
}

File: Application/src/main/kotlin/com/shareyourproxy/api/rx/event/RecyclerViewDatasetChangedEvent.kt
package com.shareyourproxy.api.rx.event


import com.shareyourproxy.app.adapter.BaseRecyclerViewAdapter
import com.shareyourproxy.util.Enumerations

/**
 * Recycler view has changed data set.
 */
internal class RecyclerViewDatasetChangedEvent(val adapter: BaseRecyclerViewAdapter, val viewState: Enumerations.ViewState)
